src/components/WidgetHolder/WidgetHolder.tsx

```diff
@@ -49,36 +82,46 @@ const WidgetHolder = () => {
 		closeAddNewWidgetModal();
 	};
 
+	const moveWidget = (fromIndex: number, toIndex: number) => {
+		const updatedWidgets = [...widgetsData];
+		const [movedWidget] = updatedWidgets.splice(fromIndex, 1);
+		updatedWidgets.splice(toIndex, 0, movedWidget);
+		setWidgetsData(updatedWidgets);
+	};
+
 	return (
-		<div className={styles.widgetHolderWrapper}>
-			<div className={styles.widgetHolder}>
-				{widgetsData.map((widgetData, index) => {
-					return (
-						<div key={index} className={styles.widget}>
-							{getWidgetDOM(widgetData, index)}
+		<DndProvider backend={HTML5Backend}>
```

Comment:
This DndProvider could be wrapped around widgetsData itself as we need drag and drop only in that region